.github/workflows/release.yml

```diff
@@ -2,78 +2,163 @@ name: Publish Build
+        run: |
+          # Cập nhật package.json
+          echo "Updating package.json with new version: ${{ env.NEW_VERSION }}"
+          sed -i "s/\"version\": \".*\"/\"version\": \"${{ env.NEW_VERSION }}\"/" package.json
+          git config user.name "GitHub Actions"
+          git config user.email "actions@github.com"
+          git add package.json
+          git commit -m "Release: v${{ env.NEW_VERSION }}"
+          git push origin ${{ env.BRANCH }} || echo "No changes to push"
 
-      - name: Get current version
-        id: get_version
-        run: echo "VERSION=$(node -p -e "require('./package.json').version")" >> $GITHUB_ENV
+#      - name: Deploy Assets to Cloudflare Pages
+#        id: cloudflare_deployment
+#        uses: cloudflare/pages-action@v1
+#        with:
+#          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+#          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
+#          projectName: 'sw-chain-list-assets'
+#          gitHubToken: ${{ secrets.GH_AUTOMATION_TOKEN }}
+#          branch: ${{ github.ref_name }}
+#          directory: './packages/chain-list-assets/public'
+#          wranglerVersion: '3'
 
       - name: Create tag
         id: create_tag
-        if: github.ref == 'refs/heads/master' && contains(github.event.head_commit.message, '[Stable]')
+        if: env.BUILD_CONDITION == 'true'
         run: |
           git config --global user.name 'github-actions[bot]'
           git config --global user.email 'github-actions[bot]@users.noreply.github.com'
-          git tag -a "v${{ env.VERSION }}" -m "Release stable version ${{ env.VERSION }}"
-          git push origin "v${{ env.VERSION }}"
+          git tag -a "v${{ env.NEW_VERSION }}" -m "Release stable version ${{ env.NEW_VERSION }}"
+          git push origin "v${{ env.NEW_VERSION }}"
+
+      - name: Post comment on PR
+        if: env.PR_NUMBER > 0 && steps.install_and_build.outcome == 'success' && env.BUILD_CONDITION == 'true'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Chuẩn bị nội dung bình luận
+          url="${{ github.event.pull_request._links.html.href}}/commits/${{github.event.after}}"
+          MSG=$(git log --format=%B -n 1 ${{ github.event.after }} | grep -v '^#')
+          COMMENT="Release for [$MSG]($url)\nNpm Version: [${{ env.NEW_VERSION }}](https://www.npmjs.com/package/${{env.PACKAGE_NAME}}/v/${{ env.NEW_VERSION }})"
+
+          # Gửi bình luận tới PR
+          curl -X POST \
+               -H "Authorization: token $GITHUB_TOKEN" \
+               -H "Accept: application/vnd.github.v3+json" \
+               -d "{\"body\": \"$COMMENT\"}" \
+               "${{ github.event.pull_request._links.comments.href}}" || echo "Failed to post comment on PR"
 
       - name: Notify to Discord
         uses: sarisia/actions-status-discord@v1
-        if: always()
+        if: always() && env.BUILD_CONDITION == 'true'
```

Comment:
Remove this condition => if failed in any task, it will update it will auto notified